src/main/java/org/lightcouch/internal/URIBuilder.java

```diff
@@ -54,17 +54,33 @@ public static URIBuilder buildUri(URI uri) {
 
 	public URI build() {
 		try {
+			StringBuilder queryBuilder = new StringBuilder(encodeQuery(query.toString()));
+
 			for (int i = 0; i < qParams.size(); i++) {
 				String amp = (i != qParams.size() - 1) ? "&" : "";
-				query.append(qParams.get(i) + amp);
+				String param;
+				String queryParam = qParams.get(i);
+				int firstEqualsIndex = queryParam.indexOf('=');
```

Comment:
I'm not sure if this is going to work properly. It assumes they don't want the first `=` encoding but they want all of the rest encoded. Why don't we investigate removing the `public URIBuilder query(String query)` flavour and then it will force API users to give their keys and values separately. If this isn't possible we should keep this flavour only for internal use and/or with a health warning to potential users.